airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -173,14 +175,26 @@ def get_sqlalchemy_engine(self, engine_kwargs=None):
         :return: the created engine.
         """
         connection = self.get_connection(self.gcp_conn_id)
-        extras = connection.extra_dejson
-        credentials_path = extras.get("extra__google_cloud_platform__key_path", None)
-        if credentials_path is None:
-            raise AirflowException(
-                "For now, we only support instantiating SQLAlchemy engine by"
-                " using extra__google_cloud_platform__key_path"
+        if os.environ.get("GOOGLE_APPLICATION_CREDENTIALS") is not None:
+            return create_engine(self.get_uri(), **engine_kwargs)
+        elif connection.extra_dejson.get("extra__google_cloud_platform__key_path"):
+            credentials_path = connection.extra_dejson['extra__google_cloud_platform__key_path']
+            return create_engine(self.get_uri(), credentials_path=credentials_path, **engine_kwargs)
+        elif connection.extra_dejson.get("extra__google_cloud_platform__keyfile_dict"):
+            credential_file_content = json.loads(
+                connection.extra_dejson["extra__google_cloud_platform__keyfile_dict"]
             )
-        return create_engine(self.get_uri(), credentials_path=credentials_path, **engine_kwargs)
+            self.log.info("Saving credentials to %s", self.credentials_path)
+            with open(self.credentials_path, "w") as file:
+                json.dump(credential_file_content, file)
+            return create_engine(self.get_uri(), credentials_path=self.credentials_path, **engine_kwargs)
```

Comment:
This is a bit dangerous. What do you think about passing this credential using `credential_info` parameter?
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/f1a249e30050d1d92e4906b4676a6fa06faff1e2/sqlalchemy_bigquery/_helpers.py#L31
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/f1a249e30050d1d92e4906b4676a6fa06faff1e2/sqlalchemy_bigquery/base.py#L755

Reply:
oh on it, btw forgive my ignorance (security is not my area of expertise), the reason why you think it's dangerous is because I'm creating a new file and might trigger some malicious script?

Reply:
done~